File: Week2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letter(string sen);
int count_word(string sen);
int count_sentence(string sen);
int main(void)
{
    string sentences = get_string("Enter the sentences: ");
    int letter = count_letter(sentences);
    int word = count_word(sentences);
    int sentence = count_sentence(sentences);

    double letters = letter * 1.0 / word * 100;
    double sentences_1 = sentence * 1.0 / word * 100;
    int level = round(0.0588 * letters - 0.296 * sentences_1 - 15.8);

    if (level < 1) {
        printf("Before Grade 1\n");
    } else if (level > 16) {
        printf("Grade 16+\n");
    } else {
        // Always remember \n
        printf("Grade %i\n", level);
    }

    return 0;


}
int count_letter(string sen) {
    int length = strlen(sen);
    int letter_number = 0;
    for (int i = 0; i < length; i++) {
        if (('a' <= sen[i] && sen[i] <= 'z') || ('A' <= sen[i] && sen[i] <= 'Z')) {
            letter_number++;
        }
    }
    return letter_number;
}

int count_word(string sen) {
    int word_number = 1;
    for (int i = 0, n = strlen(sen); i < n; i++) {
        if (sen[i] == ' ') {
            word_number++;
        }
    }
    return word_number;
}

int count_sentence(string sen) {
    int sentence_number = 0;
    for (int i = 0, length = strlen(sen); i < length; i++) {
        if (sen[i] == '.' || sen[i] == '!' || sen[i] == '?') {
            sentence_number++;
        }
    }
    return sentence_number;
}

File: Week6/readability.py
from cs50 import get_string
text = get_string("Text: ")
cnt_letter = 0
cnt_word = 0
cnt_sentence = 0
for i in text:
    if i.isalpha():
        cnt_letter += 1
    elif i.isspace():
        cnt_word += 1
    elif i == "!" or i == "." or i == "?":
        cnt_sentence += 1

cnt_word += 1
L = cnt_letter / cnt_word * 100
S = cnt_sentence / cnt_word * 100
level = round(0.0588 * L - 0.296 * S - 15.8)

if level >= 16:
    print(f"Grade 16+")
elif level < 1:
    print("Before Grade 1")
else:
    print(f"Grade {level}")
